src/config.rs

```diff
@@ -1,28 +1,79 @@
 use std::{
     fs::{create_dir, File},
     path::PathBuf,
+    str::FromStr,
 };
 
 use bytesize::ByteSize;
-use color_eyre::eyre::{eyre, Result};
-use libp2p_core::Multiaddr;
+use color_eyre::{
+    eyre::{eyre, Result},
+    Report,
+};
 use serde::{Deserialize, Serialize};
 use tracing::instrument;
 
 use subspace_sdk::{
-    farmer::CacheDescription,
-    node::{Role, RpcMethods},
+    farmer::{self, CacheDescription, Config as SdkFarmerConfig, Farmer},
+    node::{self, Config as SdkNodeConfig, Node},
     PublicKey,
 };
 
-use crate::utils::chain_parser;
+/// defaults for the user config file
+pub(crate) const DEFAULT_PLOT_SIZE: bytesize::ByteSize = bytesize::ByteSize::gb(1);
+pub(crate) const MIN_PLOT_SIZE: bytesize::ByteSize = bytesize::ByteSize::mib(32);
```

Comment:
This is not correct, you need to use plot size constant instead from `subspace-core-primitives`. The smallest plot is 1 sector. But it doesn't account for metadata and other things of course.

Reply:
will do